hbase-testing-util/src/main/java/org/apache/hadoop/hbase/testing/TestingHBaseClusterImpl.java

```diff
@@ -219,6 +222,33 @@ public List<ServerName> getRegionServerAddresses() {
       .map(t -> t.getRegionServer().getServerName()).collect(Collectors.toList());
   }
 
+  @Override
+  public Optional<Integer> getActiveMasterInfoPort() {
+    return Optional.ofNullable(util.getMiniHBaseCluster().getMaster())
+      .map(m -> m.getConfiguration().getInt(HConstants.MASTER_INFO_PORT, 0))
+      .filter(port -> port > 0);
+  }
+
+  @Override
+  public Optional<Integer> getActiveNameNodeInfoPort() {
+    return Stream.of(util.getDFSCluster().getNameNodeInfos()).map(i -> i.nameNode)
+      .filter(NameNode::isActiveState).map(nn -> nn.getHttpAddress().getPort())
```

Comment:
Another issue here is that HDFS can be configured to *not* start the http server, but only https server,

```java
set("dfs.http.policy", "HTTPS_ONLY")
set("ssl.server.keystore.location", "/tmp/keystore.jks")
set("ssl.server.keystore.password", "password")
```

and in that case `nn.getHttpAddress()` returns `null`, and we have to use the return value of `nn.getHttpsAddress()` instead.

But if we do so, it's now up to the user has to figure out which protocol is being used. So I think we should return the full address such as `http://localhost:9999` or `https://localhost:9999`.